From fc4532d0d3a58fd55cf3b379fcec9de9a79e8b59 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Wed, 20 Dec 2023 21:30:00 +0200
Subject: [PATCH] tests: Add more time for the Presence Announcements to be
 received

The previous timeout allowed only a single chirping round, and sometimes
not even that full round. That resulted in test failures for cases where
there was not really any incorrect behavior.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 tests/hwsim/test_dpp.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/hwsim/test_dpp.py b/tests/hwsim/test_dpp.py
index 909d63c92..304753316 100644
--- a/tests/hwsim/test_dpp.py
+++ b/tests/hwsim/test_dpp.py
@@ -5841,7 +5841,7 @@ def run_dpp_controller_relay(dev, apdev, params, chirp=False, discover=False,
             raise Exception("DPP_CHIRP failed")
         if duplicate:
             for i in range(10):
-                msg = relay.mgmt_rx(timeout=10)
+                msg = relay.mgmt_rx(timeout=30)
                 if msg is None:
                     raise Exception("MGMT RX wait timed out")
                 relay.request("MGMT_RX_PROCESS freq=2462 datarate=0 ssi_signal=-30 frame=" + binascii.hexlify(msg['frame']).decode())
@@ -5850,7 +5850,7 @@ def run_dpp_controller_relay(dev, apdev, params, chirp=False, discover=False,
                     relay.request("MGMT_RX_PROCESS freq=2462 datarate=0 ssi_signal=-30 frame=" + binascii.hexlify(msg['frame']).decode())
                     break
             relay.set("ext_mgmt_frame_handling", "0")
-        ev = relay.wait_event(["DPP-RX"], timeout=10)
+        ev = relay.wait_event(["DPP-RX"], timeout=30)
         if ev is None:
             raise Exception("Presence Announcement not seen")
         if "type=13" not in ev:
-- 
2.43.0

